packages/core/src/App/Containers/AccountSignupModal/account-signup-modal.jsx

```diff
@@ -54,13 +55,22 @@ const validateSignup = (values, residence_list) => {
     return errors;
 };
 
-const AccountSignup = ({ enableApp, isModalVisible, clients_country, onSignup, residence_list }) => {
+const AccountSignup = ({
+    enableApp,
+    isModalVisible,
+    clients_country,
+    onSignup,
+    residence_list,
+    isEuCountrySelected,
+}) => {
     const { is_dashboard } = React.useContext(PlatformContext);
     const [api_error, setApiError] = React.useState(false);
     const [is_loading, setIsLoading] = React.useState(true);
     const [country, setCountry] = React.useState('');
     const [pw_input, setPWInput] = React.useState('');
     const [has_valid_residence, setHasValidResidence] = React.useState(false);
+    const [selected_residence, setSelectedResidence] = React.useState('');
+    const [is_eu_resident, setIsEuResident] = React.useState(false);
```

Comment:
```suggestion
    const [selected_residence, setSelectedResidence] = React.useState('');
    const [is_eu_resident, setIsEuResident] = React.useState(false);
```
These two states could be one, instead of using `has_valid_residence` check if `selected_residence` has a value.

Reply:
Fixed